agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaScreen.kt

```diff
@@ -2,18 +2,35 @@ package com.gkp.agenda.presentation
 
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Button
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.sp
+import org.koin.androidx.compose.koinViewModel
 
 @Composable
-fun AgendaScreen() {
+fun AgendaScreen(
+    onLogout: () -> Unit
+) {
+    val viewModel = koinViewModel<AgendaViewModel>()
+
     Box(
         modifier = Modifier.fillMaxSize(),
-        contentAlignment = Alignment.Center
+
     ) {
-        Text(text = "Agenda", fontSize = 40.sp)
+        Button( modifier = Modifier.align(Alignment.TopEnd), onClick = {
+            onLogout()
+            viewModel.logout()
+        }
+        ) {
+            Text(text = "Logout")
+        }
+        Text(
+            text = "Agenda",
+            modifier = Modifier.align(Alignment.Center)
+            ,fontSize = 40.sp
+        )
     }
 }
```

Comment:
I'd avoid passing the ViewModel directly to composables, except for the root composable of each screen. This has 3 major reasons:
1. You strongly couple your composables to this specific ViewModel reference. You would not be able to reuse the composable at a place where you don't have a `AgendaViewModel`.
2. You're making it harder to have a preview, since you'll need to provide a fake ViewModel instance and fakes for everything the ViewModel needs in its constructor.
3. UI tests are also a bit harder since you'll need to provide these fakes.

What I suggest instead:
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState()) // Or Compose state alternatively
    val state = _state.asStateFlow()

    fun onAction(action: SampleAction) {
        when(action) {
            is SampleEvent.Action1 -> // Handle action1
            is SampleEvent.Action2 -> // Handle action2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    SampleScreenRoot(navController)
}

// Your screen composable
@Composable
fun SampleScreenRoot(
    navController: NavHostController,
    viewModel: SampleViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsStateWithLifecycle()
    SampleScreen(
        state = state,
        onAction = viewModel::onAction
    )
}

@Composable
fun SampleScreen(
    state: SampleState,
    onAction: (SampleAction) -> Unit
) {
    // ...
    Button(onClick = { onAction(SampleAction.ClickedButtonX) } {
        Text("Click me")
    }
}
```
This way, it's easy to preview and test `SampleScreen` since it doesn't depend on a ViewModel reference.

Reply:
I did implement the screen yet. it was just something to test the login functionality.